rsocket/statemachine/RSocketStateMachine.cpp

```diff
@@ -891,6 +930,12 @@ void RSocketStateMachine::writeNewStream(
     uint32_t initialRequestN,
     Payload payload,
     bool completed) {
+  if (coldResumeHandler_ && streamType != StreamType::FNF) {
```

Comment:
Please move the stream check to a separate method, this can probably be the same function that checks if a frame is relevant to resumption.